pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -97,7 +97,7 @@ func (r *DeploymentReconciler) checkDeploymentExist(client kclient.Client) (cons
 	}
 	//existed, check equivalence
 	//for HPA scaling, we should ignore Replicas of Deployment
-	ignoreFields := cmpopts.IgnoreFields(appsv1.DeploymentSpec{}, "Replicas")
+	ignoreFields := cmpopts.IgnoreFields(appsv1.DeploymentSpec{}, "Replicas", "Template.Spec.DeprecatedServiceAccount")
```

Comment:
We perform dry-run to populate the deployment spec before comparing, wondering why this field is not mutated by kubernetes.

Reply:
@yuzisun Sorry, tested with the wrong version. Works fine with dry-run update

Reply:
Thanks for the PR though :)